src/needleman_wunch.jl

```diff
@@ -3,13 +3,70 @@
 """
     nwscore(base1, base2)
 
-Your documentation here.
+    A fruitful function that takes in two Chars and calculates the Needleman Wunch score by comparing them. 
+    It returns an integer score based on the input values. If one of the bases is 'nothing', the score decreases by 1.
+    If both bases are 'nothing', score cannot be calculated an throws an ArgumentError.
+    Keyword arguments, such as 'match', 'mismatch', and 'gap' can be configured by the user.
+    Examples
+    ≡≡≡≡≡≡≡≡≡≡
+    
+    julia> nwscore('A', 'A')
+    1
+    julia> nwscore('A', 'G', mismatch = -2)
+    -2
+    julia> nwscore(nothing, 'G', gap = -3)
+    -3
 """
-function nwscore(base1::Char, base2::Char)
+
+function nwscore(base1::Char, base2::Char; match=1, mismatch=-1, gap=-1)
+    if(base1 isa Nothing && base2 isa Nothing)
```

Comment:
You seem to have removed the methods that handle the cases where one or both of the arguments are `Nothing` in favor of doing the type checking in a conditional. This is totally valid, but is not the preferred way to do things in julia.

Reply:
Thanks for pointing that out! What would be a better way to do this?

Reply:
The way it was originally set up - you have different methods:

```
function nwscore(c1::Char, c2::Char; kwargs...)
    # this method handles the 2 `Char` case
end

function nwscore(c1::Char, ::Nothing; kwargs...)
    # this method handles the first argument as Char, second arg as `nothing`
    # note that we don't need to name the second argument, since we never need it in the body
end

# this just calls the 2nd method, because it's the same but with order reversed
nwscore(::Nothing, c2::Char; kwargs...) = nwscore(c2, nothing; kwargs...)
```